### PR TITLE
[DOC] Rewrite new PCA section

### LINKED ISSUE #547: More documentation for --tedpca
### Summary

Between version **0.0.7** and **0.0.8**, the `--tedpca` parameter changed.
Is it planned to enhance the documentation explaining which _tedpca_ does what ?

### Additional Detail

The "decision tree" part in the code is proably the most difficult part 

Help in **0.0.7** :
```
Possible choices: mle, kundu, kundu-stabilize
Method with which to select components in TEDPCA
Default: “mle”
```

Help in **0.0.8** :
```
Possible choices: kundu, kundu-stabilize, mdl, aic, kic
Method with which to select components in TEDPCA. PCA decomposition with the mdl, kic and aic options is based on a Moving Average (stationary Gaussian) process and are ordered from most to least aggresive. Default=’mdl’.
Default: “mdl”
```

### Next Steps

Explain a bit more what does each decision tree ?
In particular, why "mdl" is choose to be the default one ?

### PR DESCRIPTION
**Pull request by eurunuela**
Closes #547 .

Changes proposed in this pull request:

- Remove confusing lines.
- Mention why we have three options.
- Add links to Wikipedia for each option as it's probably the easiest way for users to have an idea of what each option does.
- Add a note that states it is a data-driven approach and the results might not be perfect (that's also why we provide three options).

### CODE REVIEW

File: docs/approach.rst [hunk 1]
```diff
@@ -243,7 +249,9 @@ in a dimensionally reduced version of the dataset which is then used in the
 :ref:`TEDICA` step.
 
 .. image:: /_static/a12_pca_reduced_data.png
-
+.. _AIC: https://en.wikipedia.org/wiki/Akaike_information_criterion
+.. _KIC: https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence#Relationship_between_models_and_reality
```
Comment (jbteves) [inline docs/approach.rst:253]: I think we probably want to link to the top of the article, rather than this subsection.

**Review by jbteves — CHANGES_REQUESTED** [review]
Thanks for adding these changes @eurunuela ! I have one minor request, but it's otherwise good.

File: docs/approach.rst (hunk 1 above)
Comment (eurunuela) [reply docs/approach.rst:253]: Sure, whatever you think will be more meaningful for our users.

Comment (jbteves) [reply docs/approach.rst:253]: Is there something in particular you think would be valuable from that subsection? If so we could either add it to the docs or add a second link to it to highlight it.

Comment (eurunuela) [reply docs/approach.rst:253]: I wanted to highlight the relationship with AIC. I could link to the main page and then write a short sentence that says `you can see how it relates to AIC here.` with the link to that particular section.

Comment (jbteves) [reply docs/approach.rst:253]: That would be great!

Comment (eurunuela) [reply docs/approach.rst]: The latest commit deals with this. Thank you for the feedback!

File: docs/approach.rst [hunk 2]
```diff
@@ -219,7 +219,7 @@ to select the PCA components based on three widely-used model selection criterio
 * ``aic``: the Akaike Information Criterion (`AIC`_), which is the least aggressive option;
   i.e., returns the largest number of components.
 * ``kic``: the Kullback-Leibler Information Criterion (`KIC`_), which stands in the
-  middle in terms of aggressiveness.
+  middle in terms of aggressiveness. You can see how KIC related to AIC `here`_.
```
Comment (jbteves) [inline docs/approach.rst:222]:
```suggestion
  middle in terms of aggressiveness. You can see how KIC is related to AIC `here`_.
```

**Review by jbteves — CHANGES_REQUESTED** [review]
This looks good aside from a word skip. Once that's fixed I think this is ready to go.

File: docs/approach.rst (hunk 2 above)
Comment (eurunuela) [reply docs/approach.rst:222]: My bad. Working on too many things at the same time. Thank you once again Josh!

**Review by jbteves — APPROVED** [review]
Looks great, thanks @eurunuela !

File: docs/approach.rst [hunk 3]
```diff
@@ -206,22 +206,28 @@ Specifically, ``tedana`` offers two different approaches that perform this step.
...
 Gaussian process as the input.
-If we assume that the linear system is shift invariant, the moving average
-process is a stationary Gaussian random process.
 Simply put, this process more optimally selects the number of components for
 fMRI data following a subsampling scheme described in `Li et al (2007)`_.
-The selection of components is performed with either of the three options
-provided by ``--tedpca``:
 
-* ``aic``: the Akaike Information Criterion, which is the least aggressive option;
+The amount of selected principal components depends on the selection criteria.
```
Comment (tsalo) [inline docs/approach.rst:215]:
```suggestion
The number of selected principal components depends on the selection criteria.
```

File: docs/approach.rst [hunk 4]
```diff
@@ -206,22 +206,28 @@ Specifically, ``tedana`` offers two different approaches that perform this step.
...
+For this PCA method in particular, ``--tedpca`` provides three different options
+to select the PCA components based on three widely-used model selection criterion:
```
Comment (tsalo) [inline docs/approach.rst:217]:
```suggestion
to select the PCA components based on three widely-used model selection criteria:
```

File: docs/approach.rst [hunk 5]
```diff
@@ -206,22 +206,28 @@ Specifically, ``tedana`` offers two different approaches that perform this step.
...
   i.e., returns the largest number of components.
-* ``kic``: the Kullback-Leibler Information Criterion, which stands in the
-  middle in terms of aggressiveness.
-* ``mdl``: the Minimum Description Length, which is the most aggressive (and
-  recommended) option.
+* ``kic``: the Kullback-Leibler Information Criterion (`KIC`_), which stands in the
+  middle in terms of aggressiveness. You can see how KIC is related to AIC `here`_.
+* ``mdl``: the Minimum Description Length (`MDL`_), which is the most aggressive option;
+  i.e. returns the least number of components. This option is the **default and recommeded**
+  as we have seen it yields the most reasonable results.
```
Comment (tsalo) [inline docs/approach.rst:225]: I think this should be at the top of this list, since it's the default.

File: docs/approach.rst [hunk 6]
```diff
@@ -206,22 +206,28 @@ Specifically, ``tedana`` offers two different approaches that perform this step.
...
+
+.. note::
+    Please, bear in mind that this is a data-driven dimensionality reduction approach. The default
+    option ``mdl`` might not yield perfect results on your data. We suggest you explore the ``kic``
+    and ``aic`` options if running ``tedana`` with ``mdl`` returns less components than expected.
 
 A more complicated approach involves applying a decision tree (similar to the
```
Comment (tsalo) [inline docs/approach.rst:232]: Could you add something like, "In addition to the moving average process-based approaches described above, we also support a decision tree-based selection method. This method, `kundu`, ..."

File: docs/approach.rst [hunk 7]
```diff
@@ -206,22 +206,28 @@ Specifically, ``tedana`` offers two different approaches that perform this step.
 The simplest approach (the default ``mdl``, ``aic`` and ``kic`` options for
 ``--tedpca``) is based on a moving average (stationary Gaussian) process
 proposed by `Li et al (2007)`_.
```
Comment (tsalo) [inline docs/approach.rst:208]:
```suggestion
proposed by `Li et al (2007)`_ and used primarily in the Group ICA of fMRI Toolbox (GIFT).
```

**Review by tsalo — CHANGES_REQUESTED** [review]
This looks great. Thank you for doing it. I have a few minor ideas on how to provide slightly more context.

**Comment by eurunuela** [comment]
> This looks great. Thank you for doing it. I have a few minor ideas on how to provide slightly more context.

Thank you @tsalo ! Those were very helpful.

File: docs/approach.rst [hunk 8]
```diff
@@ -205,29 +205,35 @@ Specifically, ``tedana`` offers two different approaches that perform this step.
...
-TE-independent (i.e., have low Rho).
+
+.. note::
+    Please, bear in mind that this is a data-driven dimensionality reduction approach. The default
+    option ``mdl`` might not yield perfect results on your data. We suggest you explore the ``kic``
+    and ``aic`` options if running ``tedana`` with ``mdl`` returns less components than expected.
+
+In addition to the moving average process-based options described above, we also support a
+decision tree-based selection method (similar to the one in the :ref:`TEDICA` section below).
+This method, involves applying a decision tree to identify and discard PCA components which,
```
Comment (tsalo) [inline docs/approach.rst:234]:
```suggestion
This method involves applying a decision tree to identify and discard PCA components which,
```

**Review by jbteves — APPROVED** [review]

**Review by tsalo — DISMISSED** [review]

**Review by jbteves — APPROVED** [review]

**Review by tsalo — APPROVED** [review]
We should really consider disabling the "dismiss stale reviews" rule. Anyway, this looks ready to merge. Once CI passes please merge and thanks again for doing this!

**Comment by eurunuela** [comment]
Thank you guys. I'm sorry I had to asked for a review every time.

**Pull request merged**